Recursively clean up child process if we hit a timeout

desertfire opened the pull request:
This recursively cleanup is needed for releasing CUDA memory.

RUN_TORCHBENCH: ALL
TORCHBENCH_BRANCH: wconstab/ltc

desertfire committed:
Recursively clean up child process if we hit a timeout

desertfire commented on lazy_tensor_core/lazy_bench.py:
```diff
@@ -675,17 +676,21 @@ def save_error(name, test, error, dir):
                         stderr=subprocess.STDOUT)
+        try:
+            outs, errs = proc.communicate(timeout=args.timeout)
         except subprocess.TimeoutExpired:
```
@Krovatkin, per our discussion in workchat, we probably want to do the clean up for all exceptions.

Krovatkin replied:
I think the timeout exception is special because it's triggered in master rather than in a child process. I'd hope if the child process catches an exception and exits normally, we should be releasing the memory we allocated in the child process.  @wconstab  wdyt?

wconstab replied:
no idea. i guess it doesn't hurt to clean up conservatively.  has this version of the PR been shown to fix the issue so far in CI torchbench?

wconstab committed:
Avoid importing models in parent process, instead just iter names

- something in torchbench import was creating a python process and
  allocating GPU memory.  Possibly a dataloader.  This was persisting
  in the background while children subprocess then tried to work
- avoid importing torchbench models in the parent process but now need
  to handle all checking (NotImplemented, etc.) in child process

wconstab committed:
Fix RC processing
